crates/nu-cli/src/completions/completer.rs

```diff
@@ -229,6 +235,74 @@ impl ReedlineCompleter for NuCompleter {
     }
 }
 
+type MatchedAlias<'a> = Vec<(&'a [u8], &'a [u8])>;
+
+// Handler the completion when giving lines contains at least one alias. (e.g: `g checkout`)
+// that `g` is an alias of `git`
+fn try_find_alias(line: &[u8], working_set: &StateWorkingSet) -> (Vec<u8>, Vec<usize>) {
+    // An vector represents the offsets of alias
+    // e.g: the offset is 2 for the alias `g` of `git`
+    let mut alias_offset = vec![];
+    let mut output = vec![];
+    if let Some(matched_alias) = search_alias(line, working_set) {
+        let mut lens = matched_alias.len();
+        for (input_vec, line_vec) in matched_alias {
+            alias_offset.push(line_vec.len() - input_vec.len());
+            output.extend(line_vec);
+            if lens > 1 {
+                output.push(b' ');
+                lens -= 1;
+            }
+        }
+    } else {
+        output = line.to_vec();
+    }
+
+    (output, alias_offset)
+}
+
+fn search_alias<'a>(input: &'a [u8], working_set: &'a StateWorkingSet) -> Option<MatchedAlias<'a>> {
+    let mut vec_names = vec![];
+    let mut vec_alias = vec![];
+    let mut pos = 0;
+    let mut is_alias = false;
+    for (index, character) in input.iter().enumerate() {
+        if *character == b' ' {
+            let range = &input[pos..index];
+            vec_names.push(range);
+            pos = index + 1;
+        }
+    }
+    // Push the rest to names vector.
+    if pos < input.len() {
+        vec_names.push(&input[pos..]);
+    }
+
+    for name in &vec_names {
+        if let Some(alias_id) = working_set.find_alias(name) {
+            let alias_span = working_set.get_alias(alias_id);
+            is_alias = true;
+            for alias in alias_span {
+                let name = working_set.get_span_contents(*alias);
+                if !name.is_empty() {
+                    vec_alias.push(name);
+                }
+            }
+        } else {
+            vec_alias.push(name);
+        }
+    }
+
+    // Zip names and alias vectors, the original inputs and its aliases mapping.
+    // e.g:(['g'], ['g','i','t'])
+    let output = vec_names.clone().into_iter().zip(vec_alias).collect();
```

Comment:
May be we can just move this `let output = vec_names.clone().into_iter().zip(vec_alias).collect();` under the `if is_alias`?

And I don't think the final clone on vec_names is really needed.